crates/cubecl-wgpu/src/compiler/base.rs

```diff
@@ -1,31 +1,113 @@
-use std::sync::Arc;
+use std::fmt::Display;
 
 use cubecl_common::ExecutionMode;
-use cubecl_core::{prelude::CompiledKernel, server::ComputeServer, Compiler, Feature};
-use cubecl_runtime::DeviceProperties;
-use wgpu::{Adapter, ComputePipeline, Device, Queue};
+use cubecl_core::{
+    prelude::{CompiledKernel, KernelDefinition},
+    server::ComputeServer,
+    Compiler, CompilerRepresentation, WgpuCompilationOptions,
+};
+use derive_more::derive::From;
 
-use crate::WgpuServer;
+use crate::{WgpuServer, WgslCompiler};
+
+use super::wgsl::ComputeShader;
+
+#[derive(Debug, Clone)]
+pub enum DynCompiler {
```

Comment:
It's weird to have the prefix `Dyn` when we use enum (static dispatch in Rust), but I get what it does. Maybe `CombinedCompiler`, `AutoCompiler` or `WgpuCompiler` would be better names.